Make commands a lot more customizable.

matz-e opened the pull request:
Remove `pset` and `argument` parameters, and instead parse the `command`
parameter for workflows. This allows to shuffle things around a little and
will be more flexible in the future. The following merge command would now
be possible:

    my_merge.py -o @outputfiles --inputs @inputfiles

While at it, pulling in datasets from DBS and splitting them on files broke
recently, fix it.

Fixes #207.

Please make sure that the following items are taken care of if needed:

- [x] Has the database format changed? (renamed or new columns, tables)
  Or did any of the project layout change? (files required to run the
  workflows)
  - Please increase the version number `VERSION` after the imports in
    `lobster.util`.  This will ensure that Lobster does not try to load old
    projects with an incompatible version.
- [x] Did the required _Work Queue_ version change?
  - Please update the *three* version numbers in `doc/install.rst`.  Adjust
    the tarball name in the `install_dependecies.sh` script, too.
- [x] Are all additional dependencies in `setup.py`?
- [x] [Mark any issues as closed either in commits or in this pull
  request.](https://help.github.com/articles/closing-issues-via-commit-messages/)

annawoodard commented on lobster/core/data/task.py:
```diff
@@ -400,8 +423,12 @@ def copy_inputs(data, config, env):
+            if config['executable'] == 'cmsRun':
+                filename = file
+            else:
+                filename = "root://cmsxrootd.fnal.gov/" + file
```
I'm always wary of adding more knobs until we know we need them, but maybe should this be configurable?

annawoodard commented on lobster/core/workflow.py:
```diff
@@ -151,11 +151,13 @@ class Workflow(Configurable):
             :class:`ValueError` will be raised otherwise.
         command : str
             Which executable to run (for non-CMSSW workflows)
```
`(for non-CMSSW workflows)` should be removed now, no?

annawoodard commented on lobster/core/workflow.py:
```diff
@@ -151,11 +151,13 @@ class Workflow(Configurable):
+
+            The executable string may contain `@args`, `@outputfiles`, and
+            `@inputfiles`, which will be replaced by unique arguments and
+            output as well as input files, respectively.
```
Consider clarifying here what the default is (i.e. so it is obvious to use `cmsRun simple_pset.py` instead of `cmsRun simple_pset.py @inputfiles`).

matz-e commented:
Thank you for the review! I hope that I've addressed your comments sufficiently. I've taken the opportunity to change the task wrapper to be able to write tests for it, too.

annawoodard commented on lobster/core/data/task.py:
```diff
@@ -183,6 +167,20 @@ def replace(xs, s, ys):
+    fakepath = '/store/user/foo/bar.root'
+    doc = xml.dom.minidom.parse(filename)
+    for e in doc.getElementsByTagName("lfn-to-pfn"):
+        if e.attributes["protocol"].value != "xrootd":
```
Since there might be fallback xrootd servers configured, we could check if `xrootd in e.attributes["protocol"].value` instead, and cycle through them in case the first one failed.

This is ultra-low priority-- I'm a big believer in not adding additional complexity unless there's a clear need/use case. So I'll open a separate issue just to document we considered it, and if the issue actually ever comes up, we could implement it.

annawoodard commented:
This is awesome. Thanks @matz-e! 👏 👏 👏
